### ISSUE TITLE
Filter/hide groups users in differnt groups

### ISSUE THREAD

**Issue by blu-IT** [nextcloud/server#9947]
Hello,

as mentioned here [https://github.com/nextcloud/server/issues/9936](https://github.com/nextcloud/server/issues/9936) before, I see the need of separating the access to serveral information, also regarding the nextcloud system to be GDPR complient in this not so unusual case. 

This is the situation: As nextcloud admin I configured two groups, let's say _employees_ and _customers_. Now I log in as a customer and I can see all other customers even the employees in the right top button with the user symbol.

Modifying the sharing configuration in the admin-section helps that members of each group can't see the members of the other groups, **but** now employees can't share files with customers anymore.
So some kind of hierarchic access management for this type of information/access would be very helpful.

In short: Members of the group _customers_ should **not see** each other and also not the list of employees, but vice versa employees should see members of the group customers and should be able to share files/folders with them. 

Regards

### Server configuration

**Operating system**: n.a

**Web server:** n.a.

**Database:** mysql 5

**PHP version:** 7.0

**Nextcloud version:** 13.0.1

**Comment by MorrisJobke** [comment]
cc @jospoortvliet @blizzz @nextcloud/sharing

**Comment by jospoortvliet** [comment]
@blu-IT a valid issue, yes. Our customers currently don't need this, though, so it will be a low priority issue to resolve. If you require it, consider becoming a customer.

**Comment by RasmusKallas** [comment]
I'd really like an option for having a group (admins) be available for everyone, for support chat and such. I'm planning on a deployment of hundreds of no-group users, in the current situation I would need to make a group for each one and add each admin to the groups, 1-by-1.

**Comment by szaimen** [comment]
As this sounds like a nice feature, the requests for this are quite low. Currently there are no plans to implement such a feature. Thus I will close this ticket for now. This does not mean we don't want this feature, but it is simply not on our roadmap for the near future. If somebody wants to implement this feature nevertheless we are happy to assist and help out.If you wish to have this feature implemented by the Nextcloud GmbH there is the option for consulting work on top of your Nextcloud Enterprise subscription to get your features implemented.
